stdlib/LibGit2/src/LibGit2.jl

```diff
@@ -972,12 +972,12 @@ function set_ssl_cert_locations(cert_loc)
           Cint(Consts.SET_SSL_CERT_LOCATIONS), cert_file, cert_dir)
 end
 
-function __init__()
-    @check ccall((:git_libgit2_init, :libgit2), Cint, ())
+function initialize()
     REFCOUNT[] = 1
```

Comment:
`Threads.atomic_cas!(REFCOUNT, 1) == 0 || return`?

Reply:
The one and only call to it does the check. If I put the call in more places I'd factor it this way.

Reply:
That check doesn't have the `atomic_cas!`. I don't particularly know why this is using thread-safe primitives right now, but that's how the code is written, so it seems like we are supposed to carry that through all of the say